File: theme-tests/ThemePreview.java
package hiastra.preview;

import java.util.List;

// Java 17+: generics, annotations, fields, methods, and records.
public final class ThemePreview {
    private static final int LIMIT = 3;
    private final String name;

    /** Store the theme name for a short preview. */
    public ThemePreview(String name) {
        this.name = name;
    }

    @Deprecated
    public String legacyLabel() {
        return name;
    }

    public record Swatch(String label, String color) {}

    public String render(List<Swatch> swatches) {
        StringBuilder result = new StringBuilder(name);
        for (int index = 0; index < swatches.size() && index < LIMIT; index++) {
            Swatch swatch = swatches.get(index);
            if (!swatch.label().isEmpty()) {
                result.append(" / ").append(swatch.label());
            }
        }
        return result.toString();
    }

    public static void main(String[] args) {
        var preview = new ThemePreview("HiAstra");
        System.out.println(preview.render(List.of(new Swatch("Mint", "#72E0B2"))));
    }
}

File: theme-tests/ThemePreview.cs
using System;
using System.Collections.Generic;

namespace HiAstra.Preview
{
    // C# 9+: properties, attributes, records, generics, and interpolation.
    public sealed class ThemePreview
    {
        private const int Limit = 3;
        public string Name { get; } = "HiAstra";
        public record Swatch(string Label, string Color);

        /// <summary>Format a few swatches for visual inspection.</summary>
        public IEnumerable<string> Render(IEnumerable<Swatch> swatches)
        {
            int index = 0;
            foreach (var swatch in swatches)
            {
                if (index >= Limit) yield break;
                yield return $"{Name}: {index++} / {swatch.Label}";
            }
        }

        [Obsolete("Use Render instead")]
        public string LegacyLabel() => Name;

        public static void Main()
        {
            var preview = new ThemePreview();
            bool enabled = true;
            var swatches = new[] { new Swatch("Mint", "#72E0B2") };
            if (enabled)
                foreach (string label in preview.Render(swatches))
                    Console.WriteLine(label);
        }
    }
}
